### ISSUE TITLE
Doesn't work on ios device 

### ISSUE THREAD

**Issue by abdullahknz** [tsayen/dom-to-image#139]
I'm using dom-to-image in ionic 2 app and this is my typescript code. When I trying in chrome it is working but in ios device not working. 

ERROR: oops, something went wrong! [object Event]
(My console log)

public save() {
    
    var node = document.getElementById('image')
    
     domtoimage.toPng(node)
    .then(function (dataUrl) {
      
        imagedata = dataUrl;
        console.log(imagedata);
       
    })
    .catch(function (error) {
      
        console.error('oops, something went wrong!', error);
    });
    this.isCropped = true;
  }

**Comment by alanhe421** [comment]
I have the same problem.
 SecurityError (DOM Exception 18): The operation is insecure.

**Comment by bsautel** [comment]
Indeed, it does not work on iOS. In [this comment](https://github.com/tsayen/dom-to-image/issues/74#issuecomment-270863365) I explained the workaround I use to make screen shots on iOS.

**Comment by abdullahknz** [comment]
Thanks for your help. But html2canvas have blurry problem  and it is not enough for me.

**Comment by bsautel** [comment]
Ok, I am using it to make some screen shots of very small cards and html2canvas renders them rather well on iOS 10. I assume you have some more advanced use-cases. Sorry, I have no other workaround for you. 😞

**Comment by abdullahknz** [comment]
Thank you for your interest 👍

**Comment by roman-karpovich** [comment]
Same problem for me. html2canvas can't correctly process svg charts made with d3.js.

**Comment by sydtpt** [comment]
Same problem...

**Comment by zeeshanjan82** [comment]
Same problem...

**Comment by tomdogxu** [comment]
Same problem...

**Comment by lgc2007** [comment]
> 谢谢你的帮助。但是html2canvas有模糊的问题，对我来说还不够。

现在有对应的解决方案吗

**Comment by endangeredsp** [comment]
I think you can achieve this by using an iframe. Place the iframe inside of the div that you want to "capture". The issue is the canvas tag not being picked up by dom to image, but if you are screenshotting a div with an iframe in it and the iframe has the actual canvas, it won't detect that canvas is inside. I'm experimenting now and will followup to let you know if it worked or not.
